.github/workflows/docs.yml

```diff
@@ -1,15 +1,14 @@
 name: SDK Type Docs Specification
 on:
-  push:
```

Comment:
This was remove because of double deployment, as it will be deployed both for all PRs and also when the PR is merged which I don't think is necessary.

Reply:
And it will be available in GH pages? How will it ensure that PR won‘t overwrite the current docs on master if there is only 1 URL for the GH pages?

Reply:
I am not sure I understand, but how it works is that any changes made through a PR will build a new version of the docs and deploy it to gh-pages. This is to ensure that whatever modification is made to the SDK will automatically be updated in the docs.

Reply:
And also, there is no docs in the master, all the docs are built and deployed to gh-pages during PR builds.

Reply:
But then if the changes in PR is deployed to the gh-pages, the docs will be available to the users, but they will not see the correct classes in the sdk because the PR is not merged to master?

Reply:
In java sdk the javadoc is also not deployed with every PR, but only on release: https://github.com/commercetools/commercetools-sdk-java-v2/blob/5bea8eca59df46a157a7358616af8d6afe759849/.github/workflows/release.yml#L105

Reply:
Well that is not how the official docs work, if we have to look at it this way, the javadoc doesn't show any deployment in the PR to see how the docs look like, I don't think we should go with that approach. 

The official docs gives you a preview link which can be inspected to see the changes before the PR is merged, I think this is a better approach than the javadocs implementation.

Reply:
> But then if the docs changes in PR is deployed to the gh-pages, the docs will be available to the users, but they will not see the correct classes in the sdk because the PR is not merged to master?

This doesn't matter, it is not as if the PR will take forever before it is merged. I think the time between raising the PR and getting it merged is relatively short to worry about this scenario.

Reply:
> Well that is not how the official docs work, if we have to look at it this way, the javadoc doesn't show any deployment in the PR to see how the docs looks like, I don't think we should go with that approach.
> 
> The official docs gives you a preview link which can be inspected to see the changes before the PR is merged, I think this is a better approach than the javadocs implementation.

I agree, so this GH pages link is for a preview only?

Reply:
> > Well that is not how the official docs work, if we have to look at it this way, the javadoc doesn't show any deployment in the PR to see how the docs looks like, I don't think we should go with that approach.
> > The official docs gives you a preview link which can be inspected to see the changes before the PR is merged, I think this is a better approach than the javadocs implementation.
> 
> I agree, so this GH pages link is for a preview only?

No, it's not only for previews, it's also the live gh-pages link. I can change the trigger to merge or release only, but doing this, we will lose our ability to see the updated docs before merging the PR (no preview deployment link) which was something I thought about before going with the PR trigger approach.

But if you still think we should deploy it on merge or release, then we can talk to the team about it to see which approach is more favourable.

Reply:
Let‘s discuss it in the tooling weekly 👍

Reply:
I think you mixed up the docs website with the ones for the SDK. The docs website publishes a preview for every PR using vercel. This is not the case for the SDK docs.

In the Java Repo we built the docs only for the main branch to see that it's not broken, but there is no deployment happening.

Deploy to the the GH pages happens only after a tag has been made or the build was triggered manually. This should be sufficient as the docs are mainly auto generated. Only manually written docs are sometimes built locally to check the look.